src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -1099,33 +1214,38 @@ public void saveResolutions(int series)
           for (int k=0; k<scaledWidth; k+=tileWidth) {
             final int xx = k;
             int width = (int) Math.min(tileWidth, scaledWidth - xx);
-            for (int i=0; i<imageCount; i++) {
-              final int plane = i;
-
-              CompletableFuture<Void> future = new CompletableFuture<Void>();
-              futures.add(future);
-              executor.execute(() -> {
-                try {
-                  processTile(series, resolution, plane, xx, yy, width, height);
-                  LOGGER.info(
-                      "Successfully processed tile; resolution={} plane={} " +
-                      "xx={} yy={} width={} height={}",
-                      resolution, plane, xx, yy, width, height);
-                  future.complete(null);
-                }
-                catch (Throwable t) {
-                  future.completeExceptionally(t);
-                  LOGGER.error(
-                    "Failure processing tile; resolution={} plane={} " +
-                    "xx={} yy={} width={} height={}",
-                    resolution, plane, xx, yy, width, height, t);
-                }
-                finally {
-                  if (pb != null) {
-                    pb.step();
+            for (int l=0; l<scaledDepth; l+=chunkDepth) {
+              final int zz = l;
+              int depth = (int) Math.min(chunkDepth, scaledDepth - zz);
+              for (int i=0; i<ctIndices[l].size(); i++) {
+                final int plane = ctIndices[l].get(i);
```

Comment:
Is there a reason to have ```ctIndices``` instead of just looping over C and T (anticipating chunking in those dimensions as well)?